integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystem.java

```diff
@@ -133,27 +133,33 @@ public int chmod(String path, @mode_t long mode) {
   /**
    * Changes the owner of an Alluxio file.
    *
-   * This operation only works when the group mapping service is shelled based and the user is
-   * registered in unix. Otherwise it errors as not implemented. This is because the input uid and
-   * gid must match the user and group in Unix.
+   * This operation only works when the shell user group translation is enabled in Alluxio-Fuse,
+   * the user and group are registered in unix, and the user belongs to the group.
+   * Otherwise it errors as not implemented. This is because the input uid and gid
+   * must match the user and group in Unix and should be valid.
    */
   @Override
   public int chown(String path, @uid_t long uid, @gid_t long gid) {
-    if (!mIsShellGroupMapping) {
-      LOG.info("Cannot change the owner of path {} because the group mapping is not shell based",
-          path);
-      // not supported if the group mapping is not shell based
+    if (!mIsUserGroupTranslation) {
+      LOG.info("Cannot change the owner of path {} "
+          + "because the shell user group translation is not enabled in Alluxio-Fuse.", path);
       return -ErrorCodes.ENOSYS();
     }
     try {
       String userName = AlluxioFuseUtils.getUserName(uid);
-      String groupName = AlluxioFuseUtils.getGroupName(uid);
+      String groupName = AlluxioFuseUtils.getGroupName(gid);
       if (userName.isEmpty()) {
         LOG.error("Failed to get user name from uid {}", uid);
         return -ErrorCodes.EFAULT();
       }
       if (groupName.isEmpty()) {
-        LOG.error("Failed to get group name from uid {}", uid);
+        LOG.error("Failed to get group name from gid {}. ", gid);
+        return -ErrorCodes.EFAULT();
+      }
+      if (!CommonUtils.getGroups(userName).contains(groupName)) {
```

Comment:
why do we require the user to be a part of the group?

Reply:
Will it be valid If we chown the user and group of a file but the user does not belong to group?

Reply:
The user field says who the first 3 permission bits are referring to, and the group field says who the middle 3 permission bits are referring to. These can be independent

Reply:
Thanks! then I don't need to check if user belongs to group.
https://unix.stackexchange.com/questions/71032/file-owner-must-belong-to-file-group
The link also confirms that.